codecov/CMakeLists.txt

```diff
@@ -13,16 +13,36 @@
 #
 # The report is generated in REPORT_DIR and can be viewed in a web browser.
 
+# If we use clang, prefer the LLVM gcov. The "normal" gcov segfaults with the
+# coverage info generated by clang, and the LLVM gcov with GCC-generated
+# coverage gives weird errors like GCOV_TAG_COUNTER_ARCS mismatch.
+set(GCOV_NAMES "gcov")
+if(CMAKE_C_COMPILER_ID MATCHES "Clang|AppleClang")
+  string(REGEX MATCH "^[0-9]+" CMAKE_C_COMPILER_VERSION_MAJOR
+               ${CMAKE_C_COMPILER_VERSION})
+  list(PREPEND GCOV_NAMES "llvm-cov-${CMAKE_C_COMPILER_VERSION_MAJOR}")
+endif()
+
+find_program(GCOV NAMES ${GCOV_NAMES})
```

Comment:
For some other tools (e.g., OpenSSL, Git, Compiler), we are reporting the version used (see [this line](https://github.com/timescale/timescaledb/blob/9ec11d8af6e56dcf994c7e70c66ae40233099230/CMakeLists.txt#L287)). It might be good to do this for `GCOV` as well to get a better understanding of which tool was chosen.

Reply:
Done:
```
....
-- Using pg_isolation_regress /home/akuzm/ts1/install/lib/postgresql/pgxs/src/test/isolation/pg_isolation_regress
-- Using lcov /usr/bin/lcov:
lcov: LCOV version 1.14
-- Using gcov /usr/bin/llvm-cov-14:
Ubuntu LLVM version 14.0.0
  
  Optimized build.
  Default target: x86_64-pc-linux-gnu
  Host CPU: alderlake
-- Generate a code coverage report using the 'coverage' target after tests have run.
....
```